addons/controllers/csi/vspherecsiconfig_utils.go

```diff
@@ -91,26 +90,21 @@ func (r *VSphereCSIConfigReconciler) mapVSphereCSIConfigToDataValuesNonParavirtu
 	dvs.VSphereCSI = &DataValuesVSphereCSI{}
 	dvs.VSphereCSI.ClusterName = cluster.Name
 
-	vsphereCluster, err := cutil.GetVSphereCluster(ctx, r.Client, cluster)
+	vsphereCluster, err := cutil.VSphereClusterNonParavirtualForCluster(ctx, r.Client, cluster)
 	if err != nil {
 		return nil, err
 	}
 	dvs.VSphereCSI.Server = vsphereCluster.Spec.Server
 	dvs.VSphereCSI.TLSThumbprint = vsphereCluster.Spec.Thumbprint
 
-	cpMachineTemplate := &capvv1beta1.VSphereMachineTemplate{}
-	if err := r.Client.Get(ctx, types.NamespacedName{
-		Namespace: cluster.Namespace,
-		Name:      cutil.ControlPlaneName(cluster.Name),
-	}, cpMachineTemplate); err != nil {
-		if apierrors.IsNotFound(err) {
-			return nil, errors.Errorf("VSphereMachineTemplate %s/%s not found", cluster.Namespace, cutil.ControlPlaneName(cluster.Name))
-		}
-		return nil, errors.Errorf("VSphereMachineTemplate %s/%s could not be fetched, error %v", cluster.Namespace, cutil.ControlPlaneName(cluster.Name), err)
+	// get the control plane machine template
+	cpMachineTemplate, err := cutil.ControlPlaneVsphereMachineTemplateNonParavirtualForCluster(ctx, r.Client, cluster)
+	if err != nil {
+		return nil, err
 	}
 
 	dvs.VSphereCSI.Datacenter = cpMachineTemplate.Spec.Template.Spec.Datacenter
-	dvs.VSphereCSI.PublicNetwork = cutil.TryParseClusterVariableString(ctx, cluster, VSphereNetworkVarName)
+	dvs.VSphereCSI.PublicNetwork = cpMachineTemplate.Spec.Template.Spec.Network.Devices[0].NetworkName
```

Comment:
need to check if Devices is empty, otherwise the controller might panic

Reply:
Done.